src/feedlistformaction.cpp

```diff
@@ -423,19 +423,27 @@ bool FeedListFormAction::process_operation(Operation op,
 	}
 	break;
 	case OP_SELECTFILTER:
-		if (filters.size() > 0) {
-			std::string newfilter;
+		if (filter_container.size() > 0) {
 			if (automatic && args->size() > 0) {
-				newfilter = (*args)[0];
+				const std::string filter_name = (*args)[0];
+				const auto filters = filter_container.get_filters();
+				const auto filter = std::find_if(filters.begin(),
+				filters.end(), [&](const FilterNameExprPair& pair) {
+					return pair.name == filter_name;
+				});
+
+				if (filter != filters.end()) {
+					apply_filter(filter->expr);
+				} else {
+					v->get_statusline().show_error(strprintf::fmt(_("No filter found with name `%s'."),
+							filter_name));
+				}
 			} else {
-				newfilter = v->select_filter(
-						filters.get_filters());
-			}
-			if (newfilter != "") {
-				apply_filter(newfilter);
+				const std::string filter_text = v->select_filter(filter_container.get_filters());
+				apply_filter(filter_text);
 			}
 		} else {
-			v->get_statusline().show_error(_("No filters defined."));
+			v->get_statusline().show_error(_("No filter_container defined."));
```

Comment:
Oops, just realized I renamed a bit too much, will fix in a moment

Reply:
Done: https://github.com/newsboat/newsboat/pull/2137/commits/7fe8eb4f6fe0a4f0da2b79d14f3a3188b5c471d6